OutSystems/outsystems-datagrid-reactive
# RGRIDT-684: Implementing API to retrieve invalid lines

This PR contains the implementation of the API to retrieve invalid lines 


### What was happening
* There was no way to retrieve information about invalid lines

### What was done
* Added new properties (hasInvalidLines, invalidLinesJSON) to the getChangesMade method.
* Created new private method inside ValidationMark that sets invalid rows into an array.
* Removed part where we were clearing metadata inside getChangesString

### Test Steps
1. Open sample
2. On ProductName column, delete the value of a cell
3. Open console and execute the following command: JSON.parse(GridAPI.GridManager.GetChangesInGrid(GridAPI.GridManager.GetActiveGrid().uniqueId))
Expected: The above method should return an object with hasInvalidLines and invalidLinesJSON  properties. The hasInvalidLines property should be true and the invalidLinesJSON should contain the dataItem of the row that has invalid mark.


### Screenshots
(prefer animated gif)


### Checklist
